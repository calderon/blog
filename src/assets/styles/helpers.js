import { css } from "styled-components"

const srOnly = css`
  position: absolute !important;
  width: 1px !important;
  height: 1px !important;
  padding: 0 !important;
  margin: -1px !important;
  overflow: hidden !important;
  clip: rect(0, 0, 0, 0) !important;
  white-space: nowrap !important;
  border: 0 !important;
`

const srOnlyFocusable = css`
  &:not(:focus) {
    ${srOnly};
  }
`

const skipLink = css`
  ${srOnly};
  ${srOnlyFocusable};
`

const _breakpoints = {
  xs: 480,
  sm: 640,
  md: 768,
  lg: 840,
  xl: 1024,
}

const devices = {
  xs: `(min-width: ${_breakpoints.xs}px)`,
  sm: `(min-width: ${_breakpoints.sm}px)`,
  md: `(min-width: ${_breakpoints.md}px)`,
  lg: `(min-width: ${_breakpoints.lg}px)`,
  xl: `(min-width: ${_breakpoints.xl}px)`,
}

export { srOnly, srOnlyFocusable, skipLink, devices }
