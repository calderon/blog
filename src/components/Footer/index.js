import React from "react"
import styled from "styled-components"

import { Container, ContainerHeader, ContainerHeading, ContainerContent, ContainerColumn } from "../Container"
import List from "../List"
import { Link } from "../Link"

const StyledExternalLinkItem = styled.li`
  font-weight: 700;

  &::after {
    content: "Â·";
    margin: 0 1rem;
  }

  &:last-child {
    &::after {
      content: "";
    }
  }
`

const ExternalLinkItem = props => {
  return (
    <StyledExternalLinkItem>
      <Link {...props} />
    </StyledExternalLinkItem>
  )
}

const ExternalLinks = props => {
  return (
    <Container as="aside" bright inline>
      <ContainerHeading>I'm also in</ContainerHeading>
      <List unstyled inlineBlock>
        <ExternalLinkItem to="https://github.com/calderon" title="github" me />

        <ExternalLinkItem
          to="https://twitter.com/dcalderon"
          title="twitter"
          me
        />

        <ExternalLinkItem
          to="http://linkedin.com/in/danielcalderon"
          title="linkedin"
          me
        />

        <ExternalLinkItem
          to="https://stackoverflow.com/users/7442990/daniel-calder%c3%b3n-portillo"
          title="stack overflow"
          me
        />
      </List>
    </Container>
  )
}


const ContactMe = props => {
  return (
    <Container as="aside" id="contact" primary>
      <ContainerHeader>
        <ContainerHeading>Contact me</ContainerHeading>
      </ContainerHeader>
      <ContainerContent columnize={3}>
        <ContainerColumn>
          <p>Take a sit. How can I help you?</p>
        </ContainerColumn>
        <ContainerColumn>Your name + Email</ContainerColumn>
        <ContainerColumn>Your message + Send your message</ContainerColumn>
      </ContainerContent>
    </Container>
  )
}


const CopyrightDisclaimer = props => {
  return (
    <Container as="footer" secondary small noHeader>
      <ContainerContent columnize={2}>
        <ContainerColumn>
          <p>
            This blog is made with{" "}
            <Link to="https://www.gatsbyjs.org/" title="gatsby" />, hosted in{" "}
            <Link to="https://github.com/calderon/blog" title="github" me /> and
            deployed using github actions.
          </p>
        </ContainerColumn>

        <ContainerColumn opacity=".6" textRight>
          <p>
            content is{" "}
            <span className="copyright">&copy; {new Date().getFullYear()} danielcalderon.dev</span>
          </p>
        </ContainerColumn>
      </ContainerContent>
    </Container>
  )
}

const Footer = props => {
  return (
    <>
      <ContactMe />
      <ExternalLinks />
      <CopyrightDisclaimer />
    </>
  )
}

export default Footer
