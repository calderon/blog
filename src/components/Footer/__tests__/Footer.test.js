import React from "react"
import { render } from "../../../../__tests__/helpers/render"

import Footer from "../"

const renderComponent = ({ ...props } = {}) => render(<Footer {...props} />)

describe("Footer", () => {
  it("renders correctly", () => {
    const footer = renderComponent()
    expect(footer).toMatchSnapshot()
  })
})
