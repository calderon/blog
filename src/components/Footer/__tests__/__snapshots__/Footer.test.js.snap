// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <aside
        class="sc-AxirZ lmvwwP"
        id="contact"
      >
        <header
          class="sc-AxiKw ihVPJd"
        >
          <h1
            class="sc-AxjAm sc-AxhCb hqTGCv heading heading--h1"
          >
            Contact me
          </h1>
        </header>
        <div
          class="sc-AxhUy jSnPkM"
        >
          <div
            class="sc-AxgMl JpLPp"
          >
            <p>
              Take a sit. How can I help you?
            </p>
          </div>
          <div
            class="sc-AxgMl JpLPp"
          >
            Your name + Email
          </div>
          <div
            class="sc-AxgMl JpLPp"
          >
            Your message + Send your message
          </div>
        </div>
      </aside>
      <aside
        class="sc-AxirZ gLHtXh"
      >
        <h1
          class="sc-AxjAm sc-AxhCb hqTGCv heading heading--h1"
        >
          I'm also in
        </h1>
        <ul
          class="sc-AxheI ggEYld list"
        >
          <li
            class="sc-fzpans bihvlY"
          >
            <a
              class="sc-Axmtr cDyQjN"
              href="https://github.com/calderon"
              title="github"
            >
              github
            </a>
          </li>
          <li
            class="sc-fzpans bihvlY"
          >
            <a
              class="sc-Axmtr cDyQjN"
              href="https://twitter.com/dcalderon"
              title="twitter"
            >
              twitter
            </a>
          </li>
          <li
            class="sc-fzpans bihvlY"
          >
            <a
              class="sc-Axmtr cDyQjN"
              href="http://linkedin.com/in/danielcalderon"
              title="linkedin"
            >
              linkedin
            </a>
          </li>
          <li
            class="sc-fzpans bihvlY"
          >
            <a
              class="sc-Axmtr cDyQjN"
              href="https://stackoverflow.com/users/7442990/daniel-calder%c3%b3n-portillo"
              title="stack overflow"
            >
              stack overflow
            </a>
          </li>
        </ul>
      </aside>
      <footer
        class="sc-AxirZ iWVlHZ"
      >
        <div
          class="sc-AxhUy dIGGQW"
        >
          <div
            class="sc-AxgMl JpLPp"
          >
            <p>
              This blog is made with
               
              <a
                class="sc-Axmtr cDyQjN"
                href="https://www.gatsbyjs.org/"
                title="gatsby"
              >
                gatsby
              </a>
              , hosted in
               
              <a
                class="sc-Axmtr cDyQjN"
                href="https://github.com/calderon/blog"
                title="github"
              >
                github
              </a>
               and deployed using github actions.
            </p>
          </div>
          <div
            class="sc-AxgMl fsfdzF"
            opacity=".6"
          >
            <p>
              content is
               
              <span
                class="copyright"
              >
                © 
                2020
                 danielcalderon.dev
              </span>
            </p>
          </div>
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <aside
      class="sc-AxirZ lmvwwP"
      id="contact"
    >
      <header
        class="sc-AxiKw ihVPJd"
      >
        <h1
          class="sc-AxjAm sc-AxhCb hqTGCv heading heading--h1"
        >
          Contact me
        </h1>
      </header>
      <div
        class="sc-AxhUy jSnPkM"
      >
        <div
          class="sc-AxgMl JpLPp"
        >
          <p>
            Take a sit. How can I help you?
          </p>
        </div>
        <div
          class="sc-AxgMl JpLPp"
        >
          Your name + Email
        </div>
        <div
          class="sc-AxgMl JpLPp"
        >
          Your message + Send your message
        </div>
      </div>
    </aside>
    <aside
      class="sc-AxirZ gLHtXh"
    >
      <h1
        class="sc-AxjAm sc-AxhCb hqTGCv heading heading--h1"
      >
        I'm also in
      </h1>
      <ul
        class="sc-AxheI ggEYld list"
      >
        <li
          class="sc-fzpans bihvlY"
        >
          <a
            class="sc-Axmtr cDyQjN"
            href="https://github.com/calderon"
            title="github"
          >
            github
          </a>
        </li>
        <li
          class="sc-fzpans bihvlY"
        >
          <a
            class="sc-Axmtr cDyQjN"
            href="https://twitter.com/dcalderon"
            title="twitter"
          >
            twitter
          </a>
        </li>
        <li
          class="sc-fzpans bihvlY"
        >
          <a
            class="sc-Axmtr cDyQjN"
            href="http://linkedin.com/in/danielcalderon"
            title="linkedin"
          >
            linkedin
          </a>
        </li>
        <li
          class="sc-fzpans bihvlY"
        >
          <a
            class="sc-Axmtr cDyQjN"
            href="https://stackoverflow.com/users/7442990/daniel-calder%c3%b3n-portillo"
            title="stack overflow"
          >
            stack overflow
          </a>
        </li>
      </ul>
    </aside>
    <footer
      class="sc-AxirZ iWVlHZ"
    >
      <div
        class="sc-AxhUy dIGGQW"
      >
        <div
          class="sc-AxgMl JpLPp"
        >
          <p>
            This blog is made with
             
            <a
              class="sc-Axmtr cDyQjN"
              href="https://www.gatsbyjs.org/"
              title="gatsby"
            >
              gatsby
            </a>
            , hosted in
             
            <a
              class="sc-Axmtr cDyQjN"
              href="https://github.com/calderon/blog"
              title="github"
            >
              github
            </a>
             and deployed using github actions.
          </p>
        </div>
        <div
          class="sc-AxgMl fsfdzF"
          opacity=".6"
        >
          <p>
            content is
             
            <span
              class="copyright"
            >
              © 
              2020
               danielcalderon.dev
            </span>
          </p>
        </div>
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
